plugins/ilert/src/api/client.ts

```diff
@@ -0,0 +1,571 @@
+            text: 'Backstage Url',
+          },
+        ],
+        customDetails: {
+          userName: eventRequest.userName,
+        },
+      }),
+    };
+
+    const response = await this.fetch('/api/v1/events', init);
+    return response.responseCode === 'NEW_INCIDENT_CREATED';
+  }
+
+  async fetchUptimeMonitors(): Promise<UptimeMonitor[]> {
+    const init = {
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+    };
+
+    const response = await this.fetch('/api/v1/uptime-monitors', init);
+
+    return response;
+  }
+
+  async fetchUptimeMonitor(id: number): Promise<UptimeMonitor> {
+    const init = {
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+    };
+
+    const response: UptimeMonitor = await this.fetch(
+      `/api/v1/uptime-monitors/${id}`,
+      init,
+    );
+
+    return response;
+  }
+
+  async pauseUptimeMonitor(
+    uptimeMonitor: UptimeMonitor,
+  ): Promise<UptimeMonitor> {
+    const init = {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+      body: JSON.stringify({ ...uptimeMonitor, paused: true }),
+    };
+
+    const response = await this.fetch(
+      `/api/v1/uptime-monitors/${uptimeMonitor.id}`,
+      init,
+    );
+
+    return response;
+  }
+
+  async resumeUptimeMonitor(
+    uptimeMonitor: UptimeMonitor,
+  ): Promise<UptimeMonitor> {
+    const init = {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+      body: JSON.stringify({ ...uptimeMonitor, paused: false }),
+    };
+
+    const response = await this.fetch(
+      `/api/v1/uptime-monitors/${uptimeMonitor.id}`,
+      init,
+    );
+
+    return response;
+  }
+
+  async fetchAlertSources(): Promise<AlertSource[]> {
+    const init = {
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+    };
+
+    const response = await this.fetch('/api/v1/alert-sources', init);
+
+    return response;
+  }
+
+  async fetchAlertSource(
+    idOrIntegrationKey: number | string,
+  ): Promise<AlertSource> {
+    const init = {
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+    };
+
+    const response = await this.fetch(
+      `/api/v1/alert-sources/${idOrIntegrationKey}`,
+      init,
+    );
+
+    return response;
+  }
+
+  async fetchAlertSourceOnCalls(alertSource: AlertSource): Promise<OnCall[]> {
+    const init = {
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+      },
+    };
+
+    const response = await this.fetch(
+      `/api/v1/on-calls?policies=${
+        alertSource.escalationPolicy.id
+      }&expand=user&expand=escalationPolicy&timezone=${momentTimezone.tz.guess()}`,
```

Comment:
Could arguably make sense to return in UTC always and render client side with the right time zone? Maybe even always in UTC for that too - but it's a matter of taste. We have talked about introducing a core component that is able to take any ISO or similar time string, or even epoch numeric times, and render it in a user-controllable way.

Reply:
For now I think showing the on-call card using browser time is the best way. We can adjust this later, when the core component you’re talking about is ready :)